### ISSUE TITLE
Document the return value and any exceptions thrown by basic/publish

### ISSUE THREAD

**Issue by marcomorain** [michaelklishin/langohr#97]
Hi there 👋 

The docs for `basic/publish` don't mention the return value or exceptions thrown.

Since `basic/publish` delegates to `com.rabbitmq.client.Channel.basicPublish` I believe that the function return `nil` (since the signature is `void` in Java, and it throws IOException if an error is encountered.

**Comment by michaelklishin** [comment]
Correct. Feel free to submit a PR for languor.docs, mention [publisher confirms](http://www.rabbitmq.com/confirms.html) and then update the publishers/exchanges guide.
